sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -88,3 +92,36 @@ def test_extract_intro():
             result,
             'This is the description of the example which goes on and on')
         assert_false('second paragraph' in result)
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
+    examples_dir = tempfile.mkdtemp()
+    gallery_dir = tempfile.mkdtemp()
+
+    gallery_conf = {
+        'filename_pattern': '/plot_0',
+        'examples_dirs': examples_dir,
+        'gallery_dirs': gallery_dir,
+        'mod_example_dir': 'modules/generated',
+        'doc_module': (),
+        'reference_url': {},
+    }
+
+    code_output = '**Output**:\n\n\n  ::\n\n    output\n\n\n\n'
+    # create three files in tempdir (only one matches the pattern)
+    fnames = ['plot_0.py', 'plot_1.py', 'plot_2.py']
+    for fname in fnames:
+        with open(op.join(examples_dir, fname), 'w') as f:
+            f.write('\n'.join(CONTENT))
+            f.flush()
+        # generate rst file
+        sg.generate_file_rst(fname, gallery_dir, examples_dir, gallery_conf)
+        # read rst file and check if it contains code output
+        rst_fname = op.splitext(fname)[0] + '.rst'
+        with open(op.join(gallery_dir, rst_fname), 'r') as f:
+            rst = f.read()
+            if rst_fname == 'plot_0.rst':
```

Comment:
Maybe to be more generic, do something like `if re.search(gallery_conf['filename_pattern'], rst_fname)`